### DISCUSSION TITLE
General question about detection since moving on from ddetailer

### DISCUSSION THREAD

**Discussion by TKS4** [Bing-su/adetailer#32] [General]
I recall when you made the fork you spoke about the body model having degraded in functionality. I also confirmed that seeing as how the model couldn't recognize bodies anymore to any appreciable degree.

The biggest problem I think that happened when that occurred, is we lost this insanely good functionality https://github.com/dustysys/ddetailer/issues/4#issuecomment-1311200268

I see you've taken up this work and have expanded on it drastically above the original author's work. But I was just wondering, is it at all possible to have the functionality that was lost when the body detection model ceased to function anymore? Is there a model close enough to the functionality that would serve as a good-enough drop-in to get that precise detection functionality back some day? I see recently there was an update that talks about the person_yolov8n-seg.pt model and was wondering if that would be good enough?

Also, there is a somewhat newer substantial fork of 1111 I think maybe future development should shift there seeing as how AUTO's presence has been sparse for a few weeks now. https://github.com/vladmandic/automatic

**Comment by Bing-su** [comment]
I don't want us to forget that this extension just does the inpainting for us. I don't think the performance of the detection model matters much.
I think my segment model performs well enough with a size of 1-2% compared to ddetailer's 500MB segment model.

**Reply by TKS4** [reply]
EDIT: Oh wow, I just noticed the two body models, are they new? I didn't notice them before lol.

I'm just a little bit confused. Are you saying your extension is able to do the precision mask as detailed in this comment? https://github.com/dustysys/ddetailer/issues/4#issuecomment-1311200268

I guess I'm just a dummy, but if the answer is yes, I would be super grateful if you can give me a quick tutorial on how to set up the options so that I can get the precision masking effect. I understand how to do the typical face recognition that in-paints a square area of the face detected. What I don't know how to do is achieve the precision detection that outlines the face/body so there is no more square area around the face, but instead conforms with the shape of the face.

**Reply by Bing-su** [reply]
Instead of allowing more than 3 models to be used, it became difficult to overlap the masks of "2" outputs.
